### PR TITLE
Arbitrary Sidecar implementation + build. Closes #680.

### LINKED ISSUE #680: [Feature] Implement Arbitrary for BlobTransactionSidecar properly
### Component

consensus, eips, genesis

### Describe the feature you would like

https://github.com/alloy-rs/alloy/blob/d78e79e9b328a826d924e9732589dcc3bef64f38/crates/eips/src/eip4844/sidecar.rs#L18-L18

Arbitrary impl should produce:
* random blobs + matching commitments + proofs

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dorlneylon**
## Motivation

Enhancement. Closes #680.

## PR Checklist

- [ ] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by leruaa** [comment]
Ahhh @dorlneylon I'm working on this too as mentioned on the issue ;(

File: crates/eips/src/eip4844/sidecar.rs [hunk 1]
```diff
@@ -32,6 +29,26 @@ pub struct BlobTransactionSidecar {
     pub proofs: Vec<Bytes48>,
 }
 
+#[cfg(any(test, feature = "arbitrary"))]
+impl arbitrary::Arbitrary<'static> for BlobTransactionSidecar {
```
Comment (mattsse) [inline crates/eips/src/eip4844/sidecar.rs:33]:
```suggestion
impl<'a> arbitrary::Arbitrary<'a> for BlobTransactionSidecar {
```

File: crates/eips/src/eip4844/sidecar.rs [hunk 2]
```diff
@@ -32,6 +29,26 @@ pub struct BlobTransactionSidecar {
...
+    fn arbitrary(u: &mut Unstructured<'static>) -> arbitrary::Result<Self> {
+        let num_blobs: usize = u.arbitrary()?;
```
Comment (mattsse) [inline crates/eips/src/eip4844/sidecar.rs:35]: this should be between `1..BLOBS_PER_BLOCK`(6)

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by dorlneylon** [comment]
@mattsse I have changed what you asked for, can I have my pipeline run?

**Comment by dorlneylon** [comment]
@mattsse Can I have my pipeline run and review again? I've fixed what you left for me.

File: crates/eips/Cargo.toml [hunk 3]
```diff
@@ -12,8 +12,12 @@ repository.workspace = true
 exclude.workspace = true
```
Comment (prestwich) [inline crates/eips/Cargo.toml:12]: these changes appear non-functional. please remove them :)

File: crates/consensus/Cargo.toml [hunk 4]
```diff
@@ -41,4 +41,9 @@ std = ["alloy-eips/std", "c-kzg?/std"]
 k256 = ["alloy-primitives/k256"]
 kzg = ["dep:c-kzg", "alloy-eips/kzg", "std"]
 arbitrary = ["std", "dep:arbitrary", "alloy-eips/arbitrary"]
-serde = ["dep:serde", "alloy-primitives/serde", "dep:alloy-serde", "alloy-eips/serde"]
+serde = [
```
Comment (prestwich) [inline crates/consensus/Cargo.toml:44]: these changes appear to be non-functional, please remove them :)

File: crates/eips/src/eip4844/mod.rs [hunk 5]
```diff
@@ -9,6 +9,10 @@ pub mod env_settings;
 #[cfg(feature = "kzg")]
 pub mod trusted_setup_points;
 
+/// Module houses the builder and utils for the [EIP-4844 Blob Transaction](https://eips.ethereum.org/EIPS/eip-4844#blob-transaction)
```
Comment (prestwich) [inline crates/eips/src/eip4844/mod.rs:12]: repo convention is to say the contents and let "Module houses the" be implied by context. 

`Builder and utils for the [EIP-4844 Blob Transaction](https://eips.ethereum.org/EIPS/eip-4844#blob-transaction)`

**Review by prestwich — CHANGES_REQUESTED** [review]
only repo convention things from me

File: crates/eips/Cargo.toml (hunk 3 above)
Comment (dorlneylon) [reply crates/eips/Cargo.toml]: done, sorry. they happened coincidentally in the process.

**Comment by dorlneylon** [comment]
@mattsse @prestwich your reviews have been resolved, please check. Also I would like to get another workflow try (last, I hope.)

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/request.rs:1]: why was this changed

File: crates/rpc-types/src/eth/transaction/mod.rs
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/mod.rs:1]: same q

File: crates/eips/src/eip4844/builder.rs [hunk 6]
```diff
@@ -1,17 +1,30 @@
...
+};
+
+#[cfg(any(feature="kzg", feature = "arbitrary"))]
+use crate::eip4844::BlobTransactionSidecar;
+#[cfg(feature = "kzg")]
+use crate::eip4844::Bytes48;
+#[cfg(feature = "kzg")]
+use crate::eip4844::env_settings::EnvKzgSettings;
+#[cfg(feature = "kzg")]
+use core::mem;
```
Comment (mattsse) [inline crates/eips/src/eip4844/builder.rs:20]: please use full paths instead

**Review by mattsse — CHANGES_REQUESTED** [review]
not actually sure we want to move the builder to the eips crate,
but perhaps we reached a state where we need a standalone crate for 4844, because I worry a little that eips is getting bloated

File: crates/eips/src/eip4844/builder.rs (hunk 6 above)
Comment (dorlneylon) [reply crates/eips/src/eip4844/builder.rs]: done

File: crates/rpc-types/src/eth/transaction/mod.rs
Comment (dorlneylon) [reply crates/rpc-types/src/eth/transaction/mod.rs:1]: it was returned to the state it was.

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (dorlneylon) [reply crates/rpc-types/src/eth/transaction/request.rs:1]: it was returned to the state it was.

**Comment by dorlneylon** [comment]
@prestwich @mattsse your reviews have been taken care of

**Review by mattsse — COMMENTED** [review]
still not sure why the builder was moved, appears to be unused in eips

not convinced the belongs in the eips crate

**Comment by prestwich** [comment]
> still not sure why the builder was moved, appears to be unused in eips

i seem to remember that I asked for it to be moved. The sidecar is difficult to construct and use properly without it, and it doesn't really belong in the consensus crate either

File: crates/eips/src/eip4844/sidecar.rs [hunk 7]
```diff
@@ -3,6 +3,8 @@
 use crate::eip4844::{
     kzg_to_versioned_hash, Blob, Bytes48, BYTES_PER_BLOB, BYTES_PER_COMMITMENT, BYTES_PER_PROOF,
 };
+#[cfg(feature = "arbitrary")]
+use alloy_primitives::private::arbitrary::{Arbitrary, Result, Unstructured};
```
[the remark this answers is not in the document]
Comment (dorlneylon) [reply crates/eips/src/eip4844/sidecar.rs]: done

**Comment by dorlneylon** [comment]
@mattsse "use" of arbitrary impl has been removed, the changes were made.

**Comment by dorlneylon** [comment]
@mattsse

**Comment by dorlneylon** [comment]
@prestwich Can I get a review and a worflow run, please?..

**Comment by dorlneylon** [comment]
@mattsse

**Comment by dorlneylon** [comment]
@mattsse need your review.

**Review by mattsse — APPROVED** [review]
I see,

the builder is very lean, so I'm okay with this.

**Comment by dorlneylon** [comment]
@prestwich and also need yours review.
